fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx

```diff
@@ -200,6 +200,53 @@ export class AltitudeFormat implements DataEntryFormat<number> {
   }
 }
 
+type RadioMinimumEntry = 'NONE' | 'NO' | 'NO DH' | 'NODH' | number;
+export class RadioMinimumFormat implements DataEntryFormat<string | number> {
+  public placeholder = '-----';
+
+  public maxDigits = 5;
+
+  private minValue = 1;
+
+  private maxValue = maxCertifiedAlt;
+
+  constructor(
+    minValue: Subscribable<number> = Subject.create(1),
+    maxValue: Subscribable<number> = Subject.create(maxCertifiedAlt),
+  ) {
+    minValue.sub((val) => (this.minValue = val), true);
+    maxValue.sub((val) => (this.maxValue = val), true);
+  }
+
+  public format(value: RadioMinimumEntry) {
+    if (value === null || value === undefined) {
+      return [this.placeholder, null, 'FT'] as FieldFormatTuple;
+    } else if (value === 'NONE' || value === 'NO' || value === 'NO DH' || value === 'NODH') {
+      return [value, null, null] as FieldFormatTuple;
+    }
+    return [Number(value).toFixed(0).toString(), null, 'FT'] as FieldFormatTuple;
+  }
+
+  public async parse(input: string) {
+    if (input === '') {
+      return null;
+    }
+    if (input === 'NONE' || input === 'NO' || input === 'NO DH' || input === 'NODH') {
+      return input;
```

Comment:
It seems weird to not normalise them in the parser.

Reply:
I was gonna comment the same but assumed it needed to keep saying the same text.

Reply:
I also wanted to do that and return null, but when I did so, I came across a problem - when entering a value into the RADIO field, going to any other MFD page and then back to PERF APPR, the value go deleted, both from the field and the actual minimum variable. That's because InputField.ts sets the actual text value of the field from the specified variable of the component. The value of those strings in minimums altitude is basically `null`, so when rendering the page again, it would fall to the if which populates placeholders in the field, because the variable would be null. That's why it's done like this, to keep even the string value in the input field. If there's a better way to do this, I'm of course open to suggestions, it's just that I didn't figure out a better way to work with this.

Reply:
> It seems weird to not normalise them in the parser. Why not do that and avoid any other code having to deal with multiple different values that have the same meaning?

So what's the verdict on this one ?